### ISSUE TITLE
PowerShell Core Support Discussion

### ISSUE THREAD

**Issue by ephos** [devblackops/PasswordState#7]
On import of the module in PowerShell Core (*PowerShell-6.0.0-beta.4-win10-win2016-x64 as of now*)  the private function _SetCertPolicy throws an exception since the 'ICertificatePolicy' namespace doesn't exist in .Net Core 2.x. 

This issue was addressed by a PR which added a parameter `-SkipCertificateCheck` to `Invoke-RestMethod` and `Invoke-WebRequest`.  Currently setting this globally for PS needs to be done with `$PSDefaultParameterValues` for the two cmdlets.  It looks like there may be support for a `$SkipCertificateCheckPreference` in the future.  

[PowerShell Core\6 Issue 1945](https://github.com/PowerShell/PowerShell/issues/1945)
[PowerShell Core\6 PR 2006](https://github.com/PowerShell/PowerShell/pull/2006)

I just wanted to start a topic on support for the changes.  

## Expected Behavior / Possible Solution

My immediate thinking would be to modify the private function to do a version check against PowerShell and then continue to use the ICertificatePolicy method for Windows PowerShell 5 and before and then if its core for the time being set the default parameter value with $PSDefaultParameterValues.  

In my example I check on the PSEdition property but it could check the version too, I admittedly don't know the best long term option for the check but this seems to work for the time being.

```powershell
function _SetCertPolicy
{
    # Allow untrusted SSL certs
    if ($PSVersionTable.PSEdition -eq 'Core')
    {
        $PSDefaultParameterValues.Add("Invoke-RestMethod:SkipCertificateCheck", $true)
        $PSDefaultParameterValues.Add("Invoke-WebRequest:SkipCertificateCheck", $true)
    }
    else
    {
        Add-Type -TypeDefinition @"
        using System.Net;
        using System.Security.Cryptography.X509Certificates;
        public class TrustAllCertsPolicy : ICertificatePolicy {
            public bool CheckValidationResult(
                ServicePoint srvPoint, X509Certificate certificate,
                WebRequest request, int certificateProblem) {
                return true;
            }
        }
"@
        [System.Net.ServicePointManager]::CertificatePolicy = New-Object -TypeName TrustAllCertsPolicy
    }#ifelse
}#function
```

## Current Behavior

As stated above for the time being it will allow the module to support both Core and Desktop editions of PowerShell with a fairly minimal change.

## Steps to Reproduce (for bugs)

1. Install a PowerShell Core [beta](https://github.com/PowerShell/PowerShell).
2. Launch PowerShell Core Beta.
3. `Find-Module -Name PasswordState | Install-Module -Scope CurrentUser`
4. `Import-Module -Name PasswordState`

## Context

I know PowerShell core is still in beta and this issue isn't specific to this module as I've seen many others approach API certificates the same way but as I use the PasswordState module daily and it would be nice to start using it from core as we start to expand into running PowerShell Core on Linux OS's.  

## Your Environment

* Module version used: 
    - 1.2.1
* Operating System and PowerShell version:
   - Microsoft Windows 10 Enterprise
   - PowerShell v6.0.0-beta.4

**Comment by devblackops** [comment]
This makes sense to me @ephos. Thanks for putting in the issue. Feel free to put in a PR with the changes.

If I'm not mistaken, setting `$PSDefaultParameterValues` within the module scope will not affect the usage of `Invoke-RestMethod` and `Invoke-WebRequest` outside of the PasswordState module. I wouldn't want to change the cert checking behaviour for non-PasswordState invocations.

**Comment by ephos** [comment]
@devblackops, I'm a noob to this so let me know if I missed anything or did anything incorrectly!

**Comment by devblackops** [comment]
Works great. Thanks for the fix! Merged and published v1.2.3 to the PowerShell Gallery.
